app/src/main/java/com/apps/adrcotfas/goodtime/MainActivity.java

```diff
@@ -111,35 +100,29 @@ protected void onResume() {
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
 
-        setUpPreferences();
+        mPref = setUpPreferences();
         installCustomRingtones();
         saveCurrentStateOfSoundAndWifi();
         setUpUi();
         setupAppRater();
         setUpState(savedInstanceState);
     }
 
-    private void setUpPreferences() {
-        mPref = PreferenceManager.getDefaultSharedPreferences(this);
+    private Preferences setUpPreferences() {
+        SharedPreferences mPref = PreferenceManager.getDefaultSharedPreferences(this);
         mPref.registerOnSharedPreferenceChangeListener(this);
         mPrivatePref = getSharedPreferences("preferences_private", Context.MODE_PRIVATE);
         mPrivatePref.registerOnSharedPreferenceChangeListener(this);
 
         PreferenceManager.setDefaultValues(this, R.xml.preferences, true);
 
-        migratePreferencesIfNeeded();
-    }
+        this.mPref.migratePreferencesIfNeeded();
```

Comment:
I just removed the function completely since it's not needed anymore. I used it to avoid a crash in June but no users are using a version this old.